aptos-move/framework/src/extended_checks.rs

```diff
@@ -167,7 +167,14 @@ impl<'a> ExtendedChecker<'a> {
 
     fn is_allowed_input_struct(&self, qid: QualifiedId<StructId>) -> bool {
         let name = self.env.get_struct(qid).get_full_name_with_address();
-        matches!(name.as_str(), "0x1::string::String")
+        matches!(
```

Comment:
This seems somewhat error prone as it can get out of sync with the ALLOWED_STRUCT logic in the VM. Any clever way to share some common code?

Reply:
Makes sense. May be leave a TODO here to remind ourselves?